fili-core/src/main/java/com/yahoo/bard/webservice/druid/model/filter/BoundFilter.java

```diff
@@ -0,0 +1,179 @@
+    }
+
+    public BoundFilter withUpperBound(String upper) {
+        return new BoundFilter(getDimension(), null, upper, false, false, null);
+    }
+
+    public BoundFilter withLowerAndUpperBound(String lower, String upper) {
+        return new BoundFilter(getDimension(), lower, upper, false, false, null);
+    }
+
+    public BoundFilter withLowerBoundStrict(String lower, Boolean lowerStrict) {
+        return new BoundFilter(getDimension(), lower, null, lowerStrict, false, null);
+    }
+
+    public BoundFilter withUpperBoundStrict(String upper, Boolean upperStrict) {
+        return new BoundFilter(getDimension(), null, upper, false, upperStrict, null);
+    }
+
+    public BoundFilter withLowerBoundStrictUpperBound(String lower, String upper, Boolean lowerStrict) {
+        return new BoundFilter(getDimension(), lower, upper, lowerStrict, null, null);
+    }
+
+    public BoundFilter withLowerBoundUpperBoundStrict(String lower, String upper, Boolean upperStrict) {
+        return new BoundFilter(getDimension(), lower, upper, null, upperStrict, null);
+    }
+
+    public BoundFilter withLowerBoundStrictUpperBoundStrict(String lower, String upper, Boolean lowerStrict, Boolean upperStrict) {
+        return new BoundFilter(getDimension(), lower, upper, lowerStrict, upperStrict, null);
+    }
+    public BoundFilter withLowerBoundOrdering(String lower, Ordering ordering) {
+        return new BoundFilter(getDimension(), lower, null, false, false, ordering);
+    }
+
+    public BoundFilter withUpperBoundOrdering(String upper, Ordering ordering) {
+        return new BoundFilter(getDimension(), null, upper, false, false, ordering);
+    }
+
+    public BoundFilter withLowerAndUpperBoundOrdering(String lower, String upper, Ordering ordering) {
+        return new BoundFilter(getDimension(), lower, upper, false, false, ordering);
+    }
+
+    public BoundFilter withLowerBoundStrictOrdering(String lower, Boolean lowerStrict, Ordering ordering) {
+        return new BoundFilter(getDimension(), lower, null, lowerStrict, false, ordering);
+    }
+
+    public BoundFilter withUpperBoundStrictOrdering(String upper, Boolean upperStrict, Ordering ordering) {
+        return new BoundFilter(getDimension(), null, upper, false, upperStrict, ordering);
+    }
+
+    public BoundFilter withLowerBoundStrictUpperBoundOrdering(String lower, String upper, Boolean lowerStrict, Ordering ordering) {
+        return new BoundFilter(getDimension(), lower, upper, lowerStrict, null, ordering);
+    }
+
+    public BoundFilter withLowerBoundUpperBoundStrictOrdering(String lower, String upper, Boolean upperStrict, Ordering ordering) {
+        return new BoundFilter(getDimension(), lower, upper, null, upperStrict, ordering);
+    }
+
+    public BoundFilter withLowerBoundStrictUpperBoundStrictOrdering(String lower, String upper, Boolean lowerStrict, Boolean upperStrict, Ordering ordering) {
```

Comment:
I'm impressed by this parameter explosion... If we must go like this, could we use some sort of builder pattern?

Reply:
Updated